File: web-sources/s8-io-swgl/scene/pipes/color2/Color2Appearance.js
import { SWGL_Appearance } from "../SWGL_Appearance.js";


/**
 * 
 */
export class Color2Appearance extends SWGL_Appearance {


    /**
     * @type {Float32Array}
     */
    color;


    constructor(){
        super();
    }



    /**
     * 
     * @param {Float32Array} color 
     */
    S8_set_color(color) {
        this.color = color;
    }


    S8_render(){
    }


    S8_dispose(){
    }

}

File: mod-sources/com.s8.io.swgl/com/s8/io/swgl/scene/pipes/color2/Color2Appearance.java
package com.s8.io.swgl.scene.pipes.color2;

import com.s8.api.objects.web.WebS8Session;
import com.s8.io.swgl.SWGL_Root;
import com.s8.io.swgl.scene.pipes.SWGL_Appearance;
import com.s8.io.swgl.utilities.SWGL_Utilities;


/**
 * 
 * @author pierreconvert
 *
 */
public class Color2Appearance extends SWGL_Appearance {

	
	/**
	 * 
	 * @param branch
	 * @param typeName
	 */
	public Color2Appearance(WebS8Session branch) {
		super(branch, SWGL_Root.WEB+"scene/pipes/color2/Color2Appearance");
	}
	
	
	
	/**
	 * 
	 * @param material
	 */
	public void setColor(double... color) {
		vertex.fields().setFloat32ArrayField("color", SWGL_Utilities.toFloatArray(color));
	}
	

}
